src/actions/heap/heap.ts

```diff
@@ -136,25 +150,27 @@ export class HeapAction extends Hub.Action {
                 `Processed ${rowsProcessed} rows in ${
                   rowsProcessed / HeapAction.ROWS_PER_BATCH
                 } batch requests for envId ${envId}.`,
-              )
+              logTag)
             }
           }
         } catch (err) {
-          errors.push(err)
+          errors.push(new Error("Encountered an error onRow, error: " + getErrorMessage(err)))
         }
       },
     })
 
-    if (requestBatch.length > 0) {
-      requestPromises.push(
-        this.sendRequest(requestBatch, envId, requestUrl, heapField, errors),
-      )
-    }
-
     try {
+      const length = requestPromises.length
+      winston.info(`Confirming all ${length} requests are resolved`, logTag)
       await Promise.all(requestPromises)
     } catch (err) {
-      errors.push(err)
+      errors.push(new Error("Encountered an error in execute, error: " + getErrorMessage(err)))
+    }
+
+    if (requestBatch.length > 0) {
+      const length = requestBatch.length
+      winston.info(`Loading the remaining ${length} rows of data to heap`, logTag)
+      await this.sendRequest(requestBatch, envId, requestUrl, heapField, errors)
     }
```

Comment:
if we call this after the above try statement, I don't think that this error won't be caught in the `await Promise.all(requestPromises)`